kong/router/atc.lua

```diff
@@ -402,56 +403,77 @@ function _M:select(req_method, req_uri, req_host, req_scheme,
 
   local host, port = split_host_port(req_host)
 
-  for _, field in ipairs(self.fields) do
+  for i = 1, self.fields_n do
+    local field = self.fields[i]
+
     if field == "http.method" then
       assert(c:add_value(field, req_method))
+      goto continue
```

Comment:
what are the reasons for if/elseif versus if+continue?

Reply:
It makes the code more clean, easy to understand that we will process the next field immediately.

In my opinion, `else`/`elseif` is a bad pattern, it will force the reader try to find the un existed logic.

Reply:
#11278 is also a good example, in which we replace `else`/`elseif` by `return`, same as `goto continue`

Reply:
> In my opinion, else/elseif is a bad pattern

As far as coding styles go, this is far from a consensus opinion. An `if`/`elseif`/`elseif`/`elseif`/`else` block allows the reader to spot easily that the entire logic is mutually-exclusive; that is, it signals that the intent is akin to a "switch/case", by merely scanning your eyes through the structure of the outer scope.

Splitting those mutually-exclusive tests into seemingly-independent-but-not-really `if` blocks with `goto continue` entries at the bottom of each makes you have to look inside each case, only to realize there's a common pattern in the whole.

Reply:
OK, I will revert the change.